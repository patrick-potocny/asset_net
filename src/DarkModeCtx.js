import { createContext } from "react";

export const DarkModeCtx = createContext()