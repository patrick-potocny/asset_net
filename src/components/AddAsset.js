import React, { useContext, useEffect, useRef, useState } from "react";
import PropTypes from "prop-types";
import lessThan from "../assets/images/less-than.svg";
import SegmentedPicker from "./SegmentedPicker";
import SearchResults from "./SearchResults";
import { DarkModeCtx } from "../DarkModeCtx";
import searchDark from "../assets/images/searchDarkMode.svg";
import searchLight from "../assets/images/searchLightMode.svg";

function AddAsset({ onClose }) {
  const { darkMode } = useContext(DarkModeCtx);
  const currentAssetType = localStorage.getItem("assetType");
  const [assetType, setAssetType] = useState(
    currentAssetType === "all_assets" ? "crypto" : currentAssetType
  );
  const [searchIcon, setSearchIcon] = useState(null);
  const [inputVal, setInputVal] = useState(false);
  const inputRef = useRef(null);
  const buttonRef = useRef(null);

  useEffect(() => {
    setSearchIcon(darkMode ? searchDark : searchLight);
  }, [darkMode]);

  useEffect(() => {
    setInputVal(false);
  }, [assetType]);

  function handleEnter(e) {
    if (e.key === "Enter") {
      buttonRef.current.click();
    }
  }

  return (
    <div className="add-asset">
      <div className="add-asset__back-btn" onClick={onClose}>
        <img src={lessThan} alt="<" />
        <span>Back</span>
      </div>
      <h2 className="add-asset__title">Add Asset</h2>
      <SegmentedPicker
        name="search-asset-type"
        callback={(val) => {
          setAssetType(val);
        }}
        controlRef={useRef()}
        options={[
          {
            label: "Crypto",
            value: "crypto",
            ref: useRef(),
          },
          {
            label: "Stocks",
            value: "stocks",
            ref: useRef(),
          },
        ]}
        defaultOption={assetType}
      />

      <div className="add-asset__search">
        <img className="icon" src={searchIcon} alt="Magnifying glass" />
        <input
          type="text"
          placeholder="Search"
          className="input"
          onKeyDown={handleEnter}
          ref={inputRef}
        />
        <button
          className="btn"
          onClick={() => {
            setInputVal(inputRef.current.value);
          }}
          ref={buttonRef}
        >
          Search
        </button>
      </div>

      {inputVal && (
        <SearchResults
          value={inputVal}
          assetType={assetType}
          onClose={onClose}
        />
      )}
    </div>
  );
}

AddAsset.propTypes = {
  onClose: PropTypes.func.isRequired,
};

export default AddAsset;
