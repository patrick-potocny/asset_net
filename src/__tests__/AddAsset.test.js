/* eslint-disable */
import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { DarkModeCtx } from "../DarkModeCtx";
import AddAsset from "../components/AddAsset";
import userEvent from "@testing-library/user-event";

jest.mock("../components/SegmentedPicker", () => () => (
  <div>Segmented Picker</div>
));
jest.mock("../components/SearchResults", () => () => <div>Search results</div>);
const onClose = jest.fn();

describe("AddAsset component", () => {
  test("should render correctly", () => {
    const { container } = render(
      <DarkModeCtx.Provider value={{ darkMode: false }}>
        <AddAsset onClose={onClose} />
      </DarkModeCtx.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test("should change icon when darkMode is true", () => {
    render(
      <DarkModeCtx.Provider value={{ darkMode: true }}>
        <AddAsset onClose={onClose} />
      </DarkModeCtx.Provider>
    );

    expect(screen.getByAltText("Magnifying glass")).toHaveAttribute(
      "src",
      "searchDarkMode.svg"
    );
  });

  test("should call onClose when back button is clicked", () => {
    render(
      <DarkModeCtx.Provider value={{ darkMode: false }}>
        <AddAsset onClose={onClose} />
      </DarkModeCtx.Provider>
    );

    userEvent.click(screen.getByText("Back"));

    expect(onClose).toHaveBeenCalled();
  });

  test("should display search results when input changes and search button is clicked", () => {
    render(
      <DarkModeCtx.Provider value={{ darkMode: false }}>
        <AddAsset onClose={onClose} />
      </DarkModeCtx.Provider>
    );

    userEvent.type(screen.getByPlaceholderText("Search"), "tsla");
    userEvent.click(screen.getByText("Search"));

    expect(screen.getByText("Search results")).toBeInTheDocument();
  });
});
