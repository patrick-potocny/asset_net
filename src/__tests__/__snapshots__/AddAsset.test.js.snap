// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAsset component should render correctly 1`] = `
<div>
  <div
    class="add-asset"
  >
    <div
      class="add-asset__back-btn"
    >
      <img
        alt="<"
        src="less-than.svg"
      />
      <span>
        Back
      </span>
    </div>
    <h2
      class="add-asset__title"
    >
      Add Asset
    </h2>
    <div>
      Segmented Picker
    </div>
    <div
      class="add-asset__search"
    >
      <img
        alt="Magnifying glass"
        class="icon"
        src="searchLightMode.svg"
      />
      <input
        class="input"
        placeholder="Search"
        type="text"
      />
      <button
        class="btn"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
